File: Standard/Entities/SubWearhouse.cs
﻿using System;
using System.Collections.Generic;

namespace Standard.Entities;

public partial class SubWearhouse
{
    public int SubId { get; set; }

    public int MainFk { get; set; }

    public int? ParentSubWearhouseId { get; set; }

    public string SubNameEn { get; set; } = null!;

    public string SubNameAr { get; set; } = null!;

    public string? SubDescriptionEn { get; set; }

    public string? SubDescriptionAr { get; set; }

    public string? SubAddressEn { get; set; }

    public string? SubAddressAr { get; set; }

    public DateTime? SubCreatedat { get; set; }

    public DateTime? SubUpdatedat { get; set; }

    public bool? Delet { get; set; }

    public virtual ICollection<SubWearhouse> InverseParentSubWearhouse { get; set; } = new List<SubWearhouse>();

    public virtual MainWearhouse MainFkNavigation { get; set; } = null!;

    public virtual SubWearhouse? ParentSubWearhouse { get; set; }

    public virtual ICollection<Permission> PermissionDestinationSubFkNavigations { get; set; } = new List<Permission>();

    public virtual ICollection<Permission> PermissionSubFkNavigations { get; set; } = new List<Permission>();

    public virtual ICollection<SubItem> SubItems { get; set; } = new List<SubItem>();
}

File: client/src/app/shared/models/subwearhouse.ts
export interface ISubWearhouse {
    subId: number
    mainFk: number
    parentSubWearhouseId: number
    subNameEn: string
    subNameAr: string
    subDescriptionEn: string
    subDescriptionAr: string
    subAddressEn: string
    subAddressAr: string
    subCreatedat: string
    subUpdatedat: string
    delet: boolean
  }

export interface Column {
  field: string;
  header: string;
}

export interface subWearhouseVM {
  subId: number;
  subName: string;
}
